Traefik plugin - Bridge panic

Hey ! Thanks again for the release of the plugin for Traefik. I just tried it & I get a panic error when I try to go on my domain & it become unreachable. As soon as I disable the middleware/plugin, it becomes reachable again. 

It may be a misconfiguration from my part (first time using a Traefik plugin). Let me know if you need more intel.

```bash
https-proxy      | time="2021-08-24T18:47:42Z" level=info msg="Configuration loaded from file: /traefik.yml"
home-assistant   | 2021-08-24 20:49:20 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: met, hacs, binary_sensor.updater
https-proxy      | plugins-storage/sources/gop-1795076060/src/github.com/darkweak/souin/plugins/traefik/vendor/github.com/darkweak/souin/rfc/bridge.go:38:21: panic
https-proxy      | plugins-storage/sources/gop-1795076060/src/github.com/darkweak/souin/plugins/traefik/vendor/github.com/darkweak/souin/rfc/bridge.go:113:37: panic
https-proxy      | plugins-storage/sources/gop-1795076060/src/github.com/darkweak/souin/plugins/traefik/main.go:182:11: panic
https-proxy      | plugins-storage/sources/gop-1795076060/src/github.com/darkweak/souin/plugins/traefik/base.go:59:6: panic
https-proxy      | plugins-storage/sources/gop-1795076060/src/github.com/darkweak/souin/plugins/traefik/main.go:178:2: panic
https-proxy      | time="2021-08-24T18:49:24Z" level=error msg="Recovered from panic in HTTP handler [108.177.64.155:39683 - /auth/token]: reflect.Set: value of type *http.Request is not assignable to type *struct { Transport http.RoundTripper; Provider interface {}; ConfigurationURL struct { TTL struct { Duration time.Duration } \"json:\\\"ttl\\\" yaml:\\\"ttl\\\"\"; Headers []string \"json:\\\"headers\\\" yaml:\\\"headers\\\"\" }; MarkCachedResponses bool; CoalescingLayerStorage *struct {}; YkeyStorage *struct { Keys map[string]struct { URL string \"json:\\\"url\\\" yaml:\\\"url\\\"\"; Headers map[string]string \"json:\\\"headers\\\" yaml:\\\"headers\\\"\" } } }" middlewareType=Recovery middlewareName=traefik-internal-recovery
https-proxy      | time="2021-08-24T18:49:24Z" level=error msg="Stack: goroutine 163 [running]:\ngithub.com/traefik/traefik/v2/pkg/middlewares/recovery.recoverFunc({0x7f81452dd8, 0x4003aea5c0}, 0x4003ae7e00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/recovery/recovery.go:46 +0x270\npanic({0x2661120, 0x4003ca3180})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4003aecd10, 0x40028adb00, 0x4003cc6810)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4003ca3180})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4003da6210, 0x4000fd3700, 0x4003cc65f0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4003ca3180})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4003da6420, 0x40028c8b00, 0x4003cc63d0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4003ca3180})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4003c7d3f0, 0x4002316400, 0x4003cc61b0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4003ca3180})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4003cce0b0, 0x4002300600, 0x4003cc5f90)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4003ca3180})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\nreflect.Value.assignTo({0x2cc6680, 0x4003cca1e0, 0x196}, {0x2dc7161, 0xb}, 0x400279d980, 0x0)\n\t/usr/local/golang/1.10.8/go/src/reflect/value.go:2797 +0x2cc\nreflect.Value.Set({0x400279d980, 0x4003cca280, 0x196}, {0x2cc6680, 0x4003cca1e0, 0x196})\n\t/usr/local/golang/1.10.8/go/src/reflect/value.go:1905 +0xa4\ngithub.com/traefik/yaegi/interp.call.func6(0x4003cce0b0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1272 +0xb38\ngithub.com/traefik/yaegi/interp.runCfg(0x4002300600, 0x4003cce0b0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4003c7d3f0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x4002316400, 0x4003c7d3f0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4003da6420)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x40028c8b00, 0x4003da6420)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4003da6210)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x4000fd3700, 0x4003da6210)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4003aecd10)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x40028adb00, 0x4003aecd10)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.genFunctionWrapper.func2.1({0x4003adf4a0, 0x2, 0x2})\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:951 +0x41c\ngithub.com/traefik/yaegi/stdlib._net_http_Handler.ServeHTTP(...)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/stdlib/go1_17_net_http.go:269\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0x400393fd40, {0x7f81452dd8, 0x4003aea5c0}, 0x4003d94000)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:38 +0x130\ngithub.com/gorilla/mux.(*Router).ServeHTTP(0x40028d40c0, {0x7f81452dd8, 0x4003aea5c0}, 0x4003ae7e00)\n\t/home/semaphore/go/pkg/mod/github.com/containous/mux@v0.0.0-20181024131434-c33f32e26898/mux.go:133 +0x268\ngithub.com/traefik/traefik/v2/pkg/middlewares/recovery.(*recovery).ServeHTTP(0x4003c37120, {0x7f81452dd8, 0x4003aea5c0}, 0x4003ae7e00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/recovery/recovery.go:32 +0x8c\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.AddOriginFields({0x7f81452e28, 0x4003aea5b8}, 0x4003ae7e00, {0x3361e20, 0x4003c37120}, 0x4003a8f980)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:55 +0x198\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0x400393fe40, {0x7f81452e28, 0x4003aea5b8}, 0x4003ae7e00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:36 +0x100\ngithub.com/traefik/traefik/v2/pkg/middlewares/requestdecorator.(*RequestDecorator).ServeHTTP(0x4003054ac0, {0x7f81452e28, 0x4003aea5b8}, 0x4003ae7d00, 0x400009f230)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/requestdecorator/request_decorator.go:47 +0x5e8\ngithub.com/traefik/traefik/v2/pkg/middlewares/requestdecorator.WrapHandler.func1.1({0x7f81452e28, 0x4003aea5b8}, 0x4003ae7d00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/requestdecorator/request_decorator.go:84 +0x70\nnet/http.HandlerFunc.ServeHTTP(0x4003c3acc0, {0x7f81452e28, 0x4003aea5b8}, 0x4003ae7d00)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/middlewares/metrics.(*metricsMiddleware).ServeHTTP(0x4003c33880, {0x7f81452dd8, 0x4003aea5a0}, 0x4003ae7d00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/metrics/metrics.go:121 +0x754\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*Handler).ServeHTTP(0x40013548a0, {0x339e7e0, 0x4002b06fc0}, 0x4003ae7c00, {0x3361da0, 0x4003c33880})\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/logger.go:227 +0x1110\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.WrapHandler.func1.1({0x339e7e0, 0x4002b06fc0}, 0x4003ae7c00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/logger.go:71 +0x50\nnet/http.HandlerFunc.ServeHTTP(0x4003c3ad00, {0x339e7e0, 0x4002b06fc0}, 0x4003ae7c00)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/server/router/tcp.(*Manager).buildEntryPointHandler.func1({0x339e7e0, 0x4002b06fc0}, 0x4003ae7c00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/server/router/tcp/router.go:194 +0x188\nnet/http.HandlerFunc.ServeHTTP(0x4003c3aea0, {0x339e7e0, 0x4002b06fc0}, 0x4003ae7c00)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/middlewares.(*HTTPHandlerSwitcher).ServeHTTP(0x400000ff50, {0x339e7e0, 0x4002b06fc0}, 0x4003ae7c00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/handler_switcher.go:23 +0x70\ngithub.com/traefik/traefik/v2/pkg/middlewares/forwardedheaders.(*XForwarded).ServeHTTP(0x40006164b0, {0x339e7e0, 0x4002b06fc0}, 0x4003ae7c00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/forwardedheaders/forwarded_header.go:183 +0xec\nnet/http.serverHandler.ServeHTTP({0x40002ca540}, {0x339e7e0, 0x4002b06fc0}, 0x4003ae7c00)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2878 +0x45c\nnet/http.(*conn).serve(0x4000c6d680, {0x33c18c0, 0x4000796390})\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:1929 +0xb54\ncreated by net/http.(*Server).Serve\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:3033 +0x4ac\n" middlewareType=Recovery middlewareName=traefik-internal-recovery
https-proxy      | plugins-storage/sources/gop-1795076060/src/github.com/darkweak/souin/plugins/traefik/vendor/github.com/darkweak/souin/rfc/bridge.go:38:21: panic
https-proxy      | plugins-storage/sources/gop-1795076060/src/github.com/darkweak/souin/plugins/traefik/vendor/github.com/darkweak/souin/rfc/bridge.go:113:37: panic
https-proxy      | plugins-storage/sources/gop-1795076060/src/github.com/darkweak/souin/plugins/traefik/main.go:182:11: panic
https-proxy      | plugins-storage/sources/gop-1795076060/src/github.com/darkweak/souin/plugins/traefik/base.go:59:6: panic
https-proxy      | plugins-storage/sources/gop-1795076060/src/github.com/darkweak/souin/plugins/traefik/main.go:178:2: panic
https-proxy      | time="2021-08-24T18:49:24Z" level=error msg="Recovered from panic in HTTP handler [66.249.93.102:40613 - /api/google_assistant]: reflect.Set: value of type *http.Request is not assignable to type *struct { Transport http.RoundTripper; Provider interface {}; ConfigurationURL struct { TTL struct { Duration time.Duration } \"json:\\\"ttl\\\" yaml:\\\"ttl\\\"\"; Headers []string \"json:\\\"headers\\\" yaml:\\\"headers\\\"\" }; MarkCachedResponses bool; CoalescingLayerStorage *struct {}; YkeyStorage *struct { Keys map[string]struct { URL string \"json:\\\"url\\\" yaml:\\\"url\\\"\"; Headers map[string]string \"json:\\\"headers\\\" yaml:\\\"headers\\\"\" } } }" middlewareName=traefik-internal-recovery middlewareType=Recovery
https-proxy      | time="2021-08-24T18:49:24Z" level=error msg="Stack: goroutine 171 [running]:\ngithub.com/traefik/traefik/v2/pkg/middlewares/recovery.recoverFunc({0x7f81452dd8, 0x4003aeab48}, 0x4003d95a00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/recovery/recovery.go:46 +0x270\npanic({0x2661120, 0x4003d0f690})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4003da6840, 0x40028adb00, 0x4003d38810)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4003d0f690})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4003da7b80, 0x4000fd3700, 0x4003d385f0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4003d0f690})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4003da7d90, 0x40028c8b00, 0x4003d383d0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4003d0f690})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4003d3a0b0, 0x4002316400, 0x4003d381b0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4003d0f690})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4003d3ad10, 0x4002300600, 0x4003d37f90)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4003d0f690})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\nreflect.Value.assignTo({0x2cc6680, 0x4003b02830, 0x196}, {0x2dc7161, 0xb}, 0x400279d980, 0x0)\n\t/usr/local/golang/1.10.8/go/src/reflect/value.go:2797 +0x2cc\nreflect.Value.Set({0x400279d980, 0x4003b028c8, 0x196}, {0x2cc6680, 0x4003b02830, 0x196})\n\t/usr/local/golang/1.10.8/go/src/reflect/value.go:1905 +0xa4\ngithub.com/traefik/yaegi/interp.call.func6(0x4003d3ad10)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1272 +0xb38\ngithub.com/traefik/yaegi/interp.runCfg(0x4002300600, 0x4003d3ad10)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4003d3a0b0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x4002316400, 0x4003d3a0b0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4003da7d90)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x40028c8b00, 0x4003da7d90)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4003da7b80)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x4000fd3700, 0x4003da7b80)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4003da6840)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x40028adb00, 0x4003da6840)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.genFunctionWrapper.func2.1({0x4003da8990, 0x2, 0x2})\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:951 +0x41c\ngithub.com/traefik/yaegi/stdlib._net_http_Handler.ServeHTTP(...)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/stdlib/go1_17_net_http.go:269\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0x400393fd40, {0x7f81452dd8, 0x4003aeab48}, 0x4003d95c00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:38 +0x130\ngithub.com/gorilla/mux.(*Router).ServeHTTP(0x40028d40c0, {0x7f81452dd8, 0x4003aeab48}, 0x4003d95a00)\n\t/home/semaphore/go/pkg/mod/github.com/containous/mux@v0.0.0-20181024131434-c33f32e26898/mux.go:133 +0x268\ngithub.com/traefik/traefik/v2/pkg/middlewares/recovery.(*recovery).ServeHTTP(0x4003c37120, {0x7f81452dd8, 0x4003aeab48}, 0x4003d95a00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/recovery/recovery.go:32 +0x8c\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.AddOriginFields({0x7f81452e28, 0x4003aeab40}, 0x4003d95a00, {0x3361e20, 0x4003c37120}, 0x4003da4700)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:55 +0x198\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0x400393fe40, {0x7f81452e28, 0x4003aeab40}, 0x4003d95a00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:36 +0x100\ngithub.com/traefik/traefik/v2/pkg/middlewares/requestdecorator.(*RequestDecorator).ServeHTTP(0x4003054ac0, {0x7f81452e28, 0x4003aeab40}, 0x4003d95900, 0x4002ddb230)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/requestdecorator/request_decorator.go:47 +0x5e8\ngithub.com/traefik/traefik/v2/pkg/middlewares/requestdecorator.WrapHandler.func1.1({0x7f81452e28, 0x4003aeab40}, 0x4003d95900)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/requestdecorator/request_decorator.go:84 +0x70\nnet/http.HandlerFunc.ServeHTTP(0x4003c3acc0, {0x7f81452e28, 0x4003aeab40}, 0x4003d95900)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/middlewares/metrics.(*metricsMiddleware).ServeHTTP(0x4003c33880, {0x7f81452dd8, 0x4003aeab38}, 0x4003d95900)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/metrics/metrics.go:121 +0x754\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*Handler).ServeHTTP(0x40013548a0, {0x339e7e0, 0x4002b07420}, 0x4003d95800, {0x3361da0, 0x4003c33880})\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/logger.go:227 +0x1110\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.WrapHandler.func1.1({0x339e7e0, 0x4002b07420}, 0x4003d95800)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/logger.go:71 +0x50\nnet/http.HandlerFunc.ServeHTTP(0x4003c3ad00, {0x339e7e0, 0x4002b07420}, 0x4003d95800)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/server/router/tcp.(*Manager).buildEntryPointHandler.func1({0x339e7e0, 0x4002b07420}, 0x4003d95800)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/server/router/tcp/router.go:194 +0x188\nnet/http.HandlerFunc.ServeHTTP(0x4003c3aea0, {0x339e7e0, 0x4002b07420}, 0x4003d95800)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/middlewares.(*HTTPHandlerSwitcher).ServeHTTP(0x400000ff50, {0x339e7e0, 0x4002b07420}, 0x4003d95800)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/handler_switcher.go:23 +0x70\ngithub.com/traefik/traefik/v2/pkg/middlewares/forwardedheaders.(*XForwarded).ServeHTTP(0x40006164b0, {0x339e7e0, 0x4002b07420}, 0x4003d95800)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/forwardedheaders/forwarded_header.go:183 +0xec\nnet/http.serverHandler.ServeHTTP({0x40002ca540}, {0x339e7e0, 0x4002b07420}, 0x4003d95800)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2878 +0x45c\nnet/http.(*conn).serve(0x4000c6d860, {0x33c18c0, 0x4000796390})\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:1929 +0xb54\ncreated by net/http.(*Server).Serve\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:3033 +0x4ac\n" middlewareName=traefik-internal-recovery middlewareType=Recovery
```

# Configuration

#### docker-compose:
```yml
services:
  https-proxy:
    build:
      context: ./traefik
      dockerfile: Dockerfile
    network_mode: host"
    labels:
      - "traefik.enable=false"


  homeassistant:
    build:
      context: ./homeassistant
      dockerfile: Dockerfile
    labels:
      # Basic configuration
      - "traefik.enable=true"
      - "traefik.http.routers.homeassistant.rule=Host(`${HOMEASSISTANT_FQDN}`)"
      - "traefik.http.routers.homeassistant.entrypoints=websecure"
      - "traefik.http.routers.homeassistant.service=homeassistant"
      - "traefik.http.routers.homeassistant.tls=true"
      - "traefik.http.routers.homeassistant.tls.certresolver=letsencrypt"
      - "traefik.http.routers.homeassistant.middlewares=http-cache,http-compress"
      - "traefik.http.services.homeassistant.loadbalancer.server.port=8123"

      # Compress middleware
      - "traefik.http.middlewares.http-compress.compress=true"

      # HTTP cache middleware
      - "traefik.http.middlewares.http-cache.plugin.souin.default_cache.headers=Authorization,Content-Type"
      - "traefik.http.middlewares.http-cache.plugin.souin.default_cache.ttl=10s"
      - "traefik.http.middlewares.http-cache.plugin.souin.log_level=debug"
```

#### Traefik v2.5.1 configuration:

```yml
################################################################
# Providers configuration backend
################################################################
providers:
  docker:
    endpoint: unix:///var/run/docker.sock
    watch: true
    exposedByDefault: false


################################################################
# Dashboard configuration
################################################################
api:
  dashboard: true
  insecure: true


################################################################
# Pilot configuration
################################################################
pilot:
  token: "xxxxxxxxxxxxxxxxxxxxxxxxxxxx"


################################################################
# Proxy HTTP cache
################################################################
experimental:
  plugins:
    souin:
      moduleName: "github.com/darkweak/souin"
      version: "v1.5.3"
```

Hmm, look like I still get the same error.  Maybe it comes from my configuration  (not valid ?) This is my first time using a plugin as a middleware so I may have messed up here.

```bash
https-proxy      | plugins-storage/sources/gop-1678234721/src/github.com/darkweak/souin/plugins/traefik/vendor/github.com/darkweak/souin/rfc/bridge.go:38:21: panic
https-proxy      | plugins-storage/sources/gop-1678234721/src/github.com/darkweak/souin/plugins/traefik/vendor/github.com/darkweak/souin/rfc/bridge.go:118:37: panic
https-proxy      | plugins-storage/sources/gop-1678234721/src/github.com/darkweak/souin/plugins/traefik/main.go:194:11: panic
https-proxy      | plugins-storage/sources/gop-1678234721/src/github.com/darkweak/souin/plugins/traefik/base.go:62:14: panic
https-proxy      | plugins-storage/sources/gop-1678234721/src/github.com/darkweak/souin/plugins/traefik/main.go:157:13: panic
https-proxy      | time="2021-09-01T18:47:15Z" level=error msg="Recovered from panic in HTTP handler [108.177.64.157:37884 - /auth/token]: reflect.Set: value of type *http.Request is not assignable to type *struct { Transport http.RoundTripper; Provider interface {}; ConfigurationURL struct { TTL struct { Duration time.Duration } \"json:\\\"ttl\\\" yaml:\\\"ttl\\\"\"; Headers []string \"json:\\\"headers\\\" yaml:\\\"headers\\\"\" }; MarkCachedResponses bool; CoalescingLayerStorage *struct {}; YkeyStorage *struct { Cache *struct { Xcache *struct { XdefaultExpiration time.Duration; Xitems map[string]struct { Object interface {}; Expiration int64 }; Xmu sync.RWMutex; XonEvicted func(string, interface {}); Xjanitor *struct { Interval time.Duration; Xstop chan bool } } }; Keys map[string]struct { URL string \"json:\\\"url\\\" yaml:\\\"url\\\"\"; Headers map[string]string \"json:\\\"headers\\\" yaml:\\\"headers\\\"\" } } }" middlewareName=traefik-internal-recovery middlewareType=Recovery
https-proxy      | time="2021-09-01T18:47:15Z" level=error msg="Stack: goroutine 77 [running]:\ngithub.com/traefik/traefik/v2/pkg/middlewares/recovery.recoverFunc({0x7f824e6f98, 0x400350c828}, 0x4001580800)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/recovery/recovery.go:46 +0x270\npanic({0x2661120, 0x40034715d0})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x400345a9a0, 0x4002e5df00, 0x4002606810)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x40034715d0})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4001a002c0, 0x40000f4500, 0x40026065f0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x40034715d0})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4001a00580, 0x4002eae500, 0x40026063d0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x40034715d0})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x40043578c0, 0x4002835b00, 0x40026061b0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x40034715d0})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4000011290, 0x4002818d00, 0x4002605f90)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x40034715d0})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\nreflect.Value.assignTo({0x2cc6680, 0x40016b28c8, 0x196}, {0x2dc7161, 0xb}, 0x4002c93740, 0x0)\n\t/usr/local/golang/1.10.8/go/src/reflect/value.go:2797 +0x2cc\nreflect.Value.Set({0x4002c93740, 0x40016b2968, 0x196}, {0x2cc6680, 0x40016b28c8, 0x196})\n\t/usr/local/golang/1.10.8/go/src/reflect/value.go:1905 +0xa4\ngithub.com/traefik/yaegi/interp.call.func6(0x4000011290)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1272 +0xb38\ngithub.com/traefik/yaegi/interp.runCfg(0x4002818d00, 0x4000011290)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x40043578c0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x4002835b00, 0x40043578c0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4001a00580)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x4002eae500, 0x4001a00580)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4001a002c0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x40000f4500, 0x4001a002c0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x400345a9a0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x4002e5df00, 0x400345a9a0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.genFunctionWrapper.func2.1({0x40013501b0, 0x2, 0x2})\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:951 +0x41c\ngithub.com/traefik/yaegi/stdlib._net_http_Handler.ServeHTTP(...)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/stdlib/go1_17_net_http.go:269\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0x4004334580, {0x7f824e6f98, 0x400350c828}, 0x4001580a00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:38 +0x130\ngithub.com/gorilla/mux.(*Router).ServeHTTP(0x40042bb8c0, {0x7f824e6f98, 0x400350c828}, 0x4001580800)\n\t/home/semaphore/go/pkg/mod/github.com/containous/mux@v0.0.0-20181024131434-c33f32e26898/mux.go:133 +0x268\ngithub.com/traefik/traefik/v2/pkg/middlewares/recovery.(*recovery).ServeHTTP(0x40042df9f0, {0x7f824e6f98, 0x400350c828}, 0x4001580800)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/recovery/recovery.go:32 +0x8c\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.AddOriginFields({0x7f824e6fe8, 0x400350c820}, 0x4001580800, {0x3361e20, 0x40042df9f0}, 0x40024a7c40)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:55 +0x198\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0x4004334680, {0x7f824e6fe8, 0x400350c820}, 0x4001580800)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:36 +0x100\ngithub.com/traefik/traefik/v2/pkg/middlewares/requestdecorator.(*RequestDecorator).ServeHTTP(0x4003634570, {0x7f824e6fe8, 0x400350c820}, 0x4001580700, 0x40016db230)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/requestdecorator/request_decorator.go:47 +0x5e8\ngithub.com/traefik/traefik/v2/pkg/middlewares/requestdecorator.WrapHandler.func1.1({0x7f824e6fe8, 0x400350c820}, 0x4001580700)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/requestdecorator/request_decorator.go:84 +0x70\nnet/http.HandlerFunc.ServeHTTP(0x40042f3340, {0x7f824e6fe8, 0x400350c820}, 0x4001580700)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/middlewares/metrics.(*metricsMiddleware).ServeHTTP(0x400433e1c0, {0x7f824e6f98, 0x400350c808}, 0x4001580700)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/metrics/metrics.go:121 +0x754\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*Handler).ServeHTTP(0x40031f9980, {0x339e7e0, 0x4002f6a000}, 0x4001580600, {0x3361da0, 0x400433e1c0})\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/logger.go:227 +0x1110\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.WrapHandler.func1.1({0x339e7e0, 0x4002f6a000}, 0x4001580600)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/logger.go:71 +0x50\nnet/http.HandlerFunc.ServeHTTP(0x40042f3380, {0x339e7e0, 0x4002f6a000}, 0x4001580600)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/server/router/tcp.(*Manager).buildEntryPointHandler.func1({0x339e7e0, 0x4002f6a000}, 0x4001580600)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/server/router/tcp/router.go:194 +0x188\nnet/http.HandlerFunc.ServeHTTP(0x40042f3520, {0x339e7e0, 0x4002f6a000}, 0x4001580600)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/middlewares.(*HTTPHandlerSwitcher).ServeHTTP(0x40000b74c0, {0x339e7e0, 0x4002f6a000}, 0x4001580600)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/handler_switcher.go:23 +0x70\ngithub.com/traefik/traefik/v2/pkg/middlewares/forwardedheaders.(*XForwarded).ServeHTTP(0x4000205270, {0x339e7e0, 0x4002f6a000}, 0x4001580600)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/forwardedheaders/forwarded_header.go:183 +0xec\nnet/http.serverHandler.ServeHTTP({0x400001e8c0}, {0x339e7e0, 0x4002f6a000}, 0x4001580600)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2878 +0x45c\nnet/http.(*conn).serve(0x40042e46e0, {0x33c18c0, 0x4000682390})\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:1929 +0xb54\ncreated by net/http.(*Server).Serve\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:3033 +0x4ac\n" middlewareName=traefik-internal-recovery middlewareType=Recovery
https-proxy      | plugins-storage/sources/gop-1678234721/src/github.com/darkweak/souin/plugins/traefik/vendor/github.com/darkweak/souin/rfc/bridge.go:38:21: panic
https-proxy      | plugins-storage/sources/gop-1678234721/src/github.com/darkweak/souin/plugins/traefik/vendor/github.com/darkweak/souin/rfc/bridge.go:118:37: panic
https-proxy      | plugins-storage/sources/gop-1678234721/src/github.com/darkweak/souin/plugins/traefik/main.go:194:11: panic
https-proxy      | plugins-storage/sources/gop-1678234721/src/github.com/darkweak/souin/plugins/traefik/base.go:62:14: panic
https-proxy      | plugins-storage/sources/gop-1678234721/src/github.com/darkweak/souin/plugins/traefik/main.go:157:13: panic
https-proxy      | time="2021-09-01T18:47:16Z" level=error msg="Recovered from panic in HTTP handler [74.125.208.19:54651 - /api/google_assistant]: reflect.Set: value of type *http.Request is not assignable to type *struct { Transport http.RoundTripper; Provider interface {}; ConfigurationURL struct { TTL struct { Duration time.Duration } \"json:\\\"ttl\\\" yaml:\\\"ttl\\\"\"; Headers []string \"json:\\\"headers\\\" yaml:\\\"headers\\\"\" }; MarkCachedResponses bool; CoalescingLayerStorage *struct {}; YkeyStorage *struct { Cache *struct { Xcache *struct { XdefaultExpiration time.Duration; Xitems map[string]struct { Object interface {}; Expiration int64 }; Xmu sync.RWMutex; XonEvicted func(string, interface {}); Xjanitor *struct { Interval time.Duration; Xstop chan bool } } }; Keys map[string]struct { URL string \"json:\\\"url\\\" yaml:\\\"url\\\"\"; Headers map[string]string \"json:\\\"headers\\\" yaml:\\\"headers\\\"\" } } }" middlewareName=traefik-internal-recovery middlewareType=Recovery
https-proxy      | time="2021-09-01T18:47:16Z" level=error msg="Stack: goroutine 128 [running]:\ngithub.com/traefik/traefik/v2/pkg/middlewares/recovery.recoverFunc({0x7f824e6f98, 0x4003634df8}, 0x400264a100)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/recovery/recovery.go:46 +0x270\npanic({0x2661120, 0x4002415bb0})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4000474000, 0x4002e5df00, 0x40026b0810)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4002415bb0})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x40005af8c0, 0x40000f4500, 0x40026b05f0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4002415bb0})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x40005afb80, 0x4002eae500, 0x40026b03d0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4002415bb0})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4001a00c60, 0x4002835b00, 0x40026b01b0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4002415bb0})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4001a018c0, 0x4002818d00, 0x40026aff90)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x2661120, 0x4002415bb0})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\nreflect.Value.assignTo({0x2cc6680, 0x400350d080, 0x196}, {0x2dc7161, 0xb}, 0x4002c93740, 0x0)\n\t/usr/local/golang/1.10.8/go/src/reflect/value.go:2797 +0x2cc\nreflect.Value.Set({0x4002c93740, 0x400350d118, 0x196}, {0x2cc6680, 0x400350d080, 0x196})\n\t/usr/local/golang/1.10.8/go/src/reflect/value.go:1905 +0xa4\ngithub.com/traefik/yaegi/interp.call.func6(0x4001a018c0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1272 +0xb38\ngithub.com/traefik/yaegi/interp.runCfg(0x4002818d00, 0x4001a018c0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4001a00c60)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x4002835b00, 0x4001a00c60)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x40005afb80)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x4002eae500, 0x40005afb80)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x40005af8c0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x40000f4500, 0x40005af8c0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4000474000)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x4002e5df00, 0x4000474000)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.genFunctionWrapper.func2.1({0x4002334600, 0x2, 0x2})\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:951 +0x41c\ngithub.com/traefik/yaegi/stdlib._net_http_Handler.ServeHTTP(...)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/stdlib/go1_17_net_http.go:269\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0x4004334580, {0x7f824e6f98, 0x4003634df8}, 0x400264a300)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:38 +0x130\ngithub.com/gorilla/mux.(*Router).ServeHTTP(0x40042bb8c0, {0x7f824e6f98, 0x4003634df8}, 0x400264a100)\n\t/home/semaphore/go/pkg/mod/github.com/containous/mux@v0.0.0-20181024131434-c33f32e26898/mux.go:133 +0x268\ngithub.com/traefik/traefik/v2/pkg/middlewares/recovery.(*recovery).ServeHTTP(0x40042df9f0, {0x7f824e6f98, 0x4003634df8}, 0x400264a100)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/recovery/recovery.go:32 +0x8c\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.AddOriginFields({0x7f824e6fe8, 0x4003634df0}, 0x400264a100, {0x3361e20, 0x40042df9f0}, 0x4003529400)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:55 +0x198\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0x4004334680, {0x7f824e6fe8, 0x4003634df0}, 0x400264a100)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:36 +0x100\ngithub.com/traefik/traefik/v2/pkg/middlewares/requestdecorator.(*RequestDecorator).ServeHTTP(0x4003634570, {0x7f824e6fe8, 0x4003634df0}, 0x400374df00, 0x4000afb230)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/requestdecorator/request_decorator.go:47 +0x5e8\ngithub.com/traefik/traefik/v2/pkg/middlewares/requestdecorator.WrapHandler.func1.1({0x7f824e6fe8, 0x4003634df0}, 0x400374df00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/requestdecorator/request_decorator.go:84 +0x70\nnet/http.HandlerFunc.ServeHTTP(0x40042f3340, {0x7f824e6fe8, 0x4003634df0}, 0x400374df00)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/middlewares/metrics.(*metricsMiddleware).ServeHTTP(0x400433e1c0, {0x7f824e6f98, 0x4003634de8}, 0x400374df00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/metrics/metrics.go:121 +0x754\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*Handler).ServeHTTP(0x40031f9980, {0x339e7e0, 0x4002f70000}, 0x400374de00, {0x3361da0, 0x400433e1c0})\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/logger.go:227 +0x1110\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.WrapHandler.func1.1({0x339e7e0, 0x4002f70000}, 0x400374de00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/logger.go:71 +0x50\nnet/http.HandlerFunc.ServeHTTP(0x40042f3380, {0x339e7e0, 0x4002f70000}, 0x400374de00)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/server/router/tcp.(*Manager).buildEntryPointHandler.func1({0x339e7e0, 0x4002f70000}, 0x400374de00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/server/router/tcp/router.go:194 +0x188\nnet/http.HandlerFunc.ServeHTTP(0x40042f3520, {0x339e7e0, 0x4002f70000}, 0x400374de00)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/middlewares.(*HTTPHandlerSwitcher).ServeHTTP(0x40000b74c0, {0x339e7e0, 0x4002f70000}, 0x400374de00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/handler_switcher.go:23 +0x70\ngithub.com/traefik/traefik/v2/pkg/middlewares/forwardedheaders.(*XForwarded).ServeHTTP(0x4000205270, {0x339e7e0, 0x4002f70000}, 0x400374de00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/forwardedheaders/forwarded_header.go:183 +0xec\nnet/http.serverHandler.ServeHTTP({0x400001e8c0}, {0x339e7e0, 0x4002f70000}, 0x400374de00)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2878 +0x45c\nnet/http.(*conn).serve(0x40008cb180, {0x33c18c0, 0x4000682390})\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:1929 +0xb54\ncreated by net/http.(*Server).Serve\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:3033 +0x4ac\n" middlewareName=traefik-internal-recovery middlewareType=Recovery
```

# Configuration
----
I've just changed to v1.5.4 for Souin & added this line in my **docker-compose.yml**.

```yaml
      # HTTP cache middleware
      - "traefik.http.middlewares.http-cache.plugin.souin.api.souin.enable=true"
      - ...
```

Hmm unfortunately I still get the error (though I think the ones related the websockets are gone now !)

```bash
https-proxy      | plugins-storage/sources/gop-3928943194/src/github.com/darkweak/souin/plugins/traefik/vendor/github.com/darkweak/souin/rfc/bridge.go:38:21: panic
https-proxy      | plugins-storage/sources/gop-3928943194/src/github.com/darkweak/souin/plugins/traefik/vendor/github.com/darkweak/souin/rfc/bridge.go:118:37: panic
https-proxy      | plugins-storage/sources/gop-3928943194/src/github.com/darkweak/souin/plugins/traefik/main.go:199:11: panic
https-proxy      | plugins-storage/sources/gop-3928943194/src/github.com/darkweak/souin/plugins/traefik/base.go:63:14: panic
https-proxy      | plugins-storage/sources/gop-3928943194/src/github.com/darkweak/souin/plugins/traefik/main.go:157:94: panic
https-proxy      | time="2021-09-05T18:53:47Z" level=error msg="Recovered from panic in HTTP handler [108.177.64.121:45106 - /auth/token]: reflect.Set: value of type *http.Request is not assignable to type *struct { Transport http.RoundTripper; Provider interface {}; ConfigurationURL struct { TTL struct { Duration time.Duration } \"json:\\\"ttl\\\" yaml:\\\"ttl\\\"\"; Headers []string \"json:\\\"headers\\\" yaml:\\\"headers\\\"\" }; MarkCachedResponses bool; CoalescingLayerStorage *struct {}; YkeyStorage *struct { Cache *struct { Xcache *struct { XdefaultExpiration time.Duration; Xitems map[string]struct { Object interface {}; Expiration int64 }; Xmu sync.RWMutex; XonEvicted func(string, interface {}); Xjanitor *struct { Interval time.Duration; Xstop chan bool } } }; Keys map[string]struct { URL string \"json:\\\"url\\\" yaml:\\\"url\\\"\"; Headers map[string]string \"json:\\\"headers\\\" yaml:\\\"headers\\\"\" } } }" middlewareName=traefik-internal-recovery middlewareType=Recovery
https-proxy      | time="2021-09-05T18:53:47Z" level=error msg="Stack: goroutine 161 [running]:\ngithub.com/traefik/traefik/v2/pkg/middlewares/recovery.recoverFunc({0x7f7f783190, 0x400224e7e0}, 0x400404ee00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/recovery/recovery.go:46 +0x270\npanic({0x26613c0, 0x40033b6710})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x400402e6e0, 0x400283b500, 0x40042065d0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x26613c0, 0x40033b6710})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x40041a40b0, 0x4000fbc700, 0x40042063b0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x26613c0, 0x40033b6710})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x40041a4210, 0x4002870800, 0x4004206190)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x26613c0, 0x40033b6710})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4003dd6580, 0x40011a0f00, 0x4004205f70)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x26613c0, 0x40033b6710})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4003dd71e0, 0x400112c100, 0x4004205d50)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x26613c0, 0x40033b6710})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\nreflect.Value.assignTo({0x2cc6e80, 0x4001da34d0, 0x196}, {0x2dc7972, 0xb}, 0x40024ca900, 0x0)\n\t/usr/local/golang/1.10.8/go/src/reflect/value.go:2797 +0x2cc\nreflect.Value.Set({0x40024ca900, 0x4001da35b0, 0x196}, {0x2cc6e80, 0x4001da34d0, 0x196})\n\t/usr/local/golang/1.10.8/go/src/reflect/value.go:1905 +0xa4\ngithub.com/traefik/yaegi/interp.call.func6(0x4003dd71e0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1272 +0xb38\ngithub.com/traefik/yaegi/interp.runCfg(0x400112c100, 0x4003dd71e0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4003dd6580)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x40011a0f00, 0x4003dd6580)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x40041a4210)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x4002870800, 0x40041a4210)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x40041a40b0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x4000fbc700, 0x40041a40b0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x400402e6e0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x400283b500, 0x400402e6e0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.genFunctionWrapper.func2.1({0x400405ecc0, 0x2, 0x2})\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:951 +0x41c\ngithub.com/traefik/yaegi/stdlib._net_http_Handler.ServeHTTP(...)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/stdlib/go1_17_net_http.go:269\ngithub.com/klauspost/compress/gzhttp.NewWrapper.func1.1({0x7f7f783190, 0x400224e7e0}, 0x400404f000)\n\t/home/semaphore/go/pkg/mod/github.com/klauspost/compress@v1.13.0/gzhttp/gzip.go:378 +0x2dc\nnet/http.HandlerFunc.ServeHTTP(0x4002505080, {0x7f7f783190, 0x400224e7e0}, 0x400404f000)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/middlewares/compress.(*compress).ServeHTTP(0x40034581c0, {0x7f7f783190, 0x400224e7e0}, 0x400404f000)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/compress/compress.go:55 +0x234\ngithub.com/traefik/traefik/v2/pkg/middlewares/tracing.(*Wrapper).ServeHTTP(0x4003612ae0, {0x7f7f783190, 0x400224e7e0}, 0x400404f000)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/tracing/wrapper.go:57 +0x130\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0x4003458200, {0x7f7f783190, 0x400224e7e0}, 0x400404f000)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:38 +0x130\ngithub.com/gorilla/mux.(*Router).ServeHTTP(0x400315f2c0, {0x7f7f783190, 0x400224e7e0}, 0x400404ee00)\n\t/home/semaphore/go/pkg/mod/github.com/containous/mux@v0.0.0-20181024131434-c33f32e26898/mux.go:133 +0x268\ngithub.com/traefik/traefik/v2/pkg/middlewares/recovery.(*recovery).ServeHTTP(0x4002eb12d0, {0x7f7f783190, 0x400224e7e0}, 0x400404ee00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/recovery/recovery.go:32 +0x8c\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.AddOriginFields({0x7f7f7831e0, 0x400224e7d8}, 0x400404ee00, {0x33628e0, 0x4002eb12d0}, 0x40040584c0)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:55 +0x198\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0x4003458300, {0x7f7f7831e0, 0x400224e7d8}, 0x400404ee00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:36 +0x100\ngithub.com/traefik/traefik/v2/pkg/middlewares/requestdecorator.(*RequestDecorator).ServeHTTP(0x4001a88008, {0x7f7f7831e0, 0x400224e7d8}, 0x400404ed00, 0x4003fef230)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/requestdecorator/request_decorator.go:47 +0x5e8\ngithub.com/traefik/traefik/v2/pkg/middlewares/requestdecorator.WrapHandler.func1.1({0x7f7f7831e0, 0x400224e7d8}, 0x400404ed00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/requestdecorator/request_decorator.go:84 +0x70\nnet/http.HandlerFunc.ServeHTTP(0x40024ffc80, {0x7f7f7831e0, 0x400224e7d8}, 0x400404ed00)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/middlewares/metrics.(*metricsMiddleware).ServeHTTP(0x4003f8fab0, {0x7f7f783190, 0x400224e7b8}, 0x400404ed00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/metrics/metrics.go:121 +0x754\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*Handler).ServeHTTP(0x4001681b00, {0x339f2c8, 0x40028ded20}, 0x400404ec00, {0x3362860, 0x4003f8fab0})\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/logger.go:227 +0x1110\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.WrapHandler.func1.1({0x339f2c8, 0x40028ded20}, 0x400404ec00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/logger.go:71 +0x50\nnet/http.HandlerFunc.ServeHTTP(0x40024ffcc0, {0x339f2c8, 0x40028ded20}, 0x400404ec00)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/server/router/tcp.(*Manager).buildEntryPointHandler.func1({0x339f2c8, 0x40028ded20}, 0x400404ec00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/server/router/tcp/router.go:194 +0x188\nnet/http.HandlerFunc.ServeHTTP(0x40024ffe80, {0x339f2c8, 0x40028ded20}, 0x400404ec00)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/middlewares.(*HTTPHandlerSwitcher).ServeHTTP(0x40000bf868, {0x339f2c8, 0x40028ded20}, 0x400404ec00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/handler_switcher.go:23 +0x70\ngithub.com/traefik/traefik/v2/pkg/middlewares/forwardedheaders.(*XForwarded).ServeHTTP(0x40000b2a50, {0x339f2c8, 0x40028ded20}, 0x400404ec00)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/forwardedheaders/forwarded_header.go:183 +0xec\nnet/http.serverHandler.ServeHTTP({0x400047e540}, {0x339f2c8, 0x40028ded20}, 0x400404ec00)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2878 +0x45c\nnet/http.(*conn).serve(0x4003fceb40, {0x33c23e0, 0x40005a0150})\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:1929 +0xb54\ncreated by net/http.(*Server).Serve\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:3033 +0x4ac\n" middlewareName=traefik-internal-recovery middlewareType=Recovery
```


# Configuration
Now using Souin plugin v1.5.6 & Traefik v2.5.2. I will try to get a pcap soon to investigate a bit more but this is related to Google APIs calls to my remote homeassistant whenever I say something. This is probably a POST request. 


Regarding the excluding regex, can it be an array or do I need to make 'one' regex rule matching all the domains/urls I want to exclude ?

Okay, just reproduced it easily using your environment. This is a bit trickier to reproduce with homeassistant as those POST requests are related to the mobile access or Google assistant integration in my case (**mobile_app** for example, which are not enabled by default so you wouldn't get those with a clean install).

Anyway, using **traefik/whoami**, if you make a simple POST request using curl for example, you should get the same errors. 

```bash
curl -X POST https://XXXXXX
curl: (92) HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1)
```

```bash
https-proxy      | plugins-storage/sources/gop-2911974483/src/github.com/darkweak/souin/plugins/traefik/vendor/github.com/darkweak/souin/rfc/bridge.go:38:21: panic
https-proxy      | plugins-storage/sources/gop-2911974483/src/github.com/darkweak/souin/plugins/traefik/vendor/github.com/darkweak/souin/rfc/bridge.go:120:37: panic
https-proxy      | plugins-storage/sources/gop-2911974483/src/github.com/darkweak/souin/plugins/traefik/main.go:199:11: panic
https-proxy      | plugins-storage/sources/gop-2911974483/src/github.com/darkweak/souin/plugins/traefik/base.go:63:14: panic
https-proxy      | plugins-storage/sources/gop-2911974483/src/github.com/darkweak/souin/plugins/traefik/main.go:157:94: panic
https-proxy      | time="2021-09-06T19:01:01Z" level=error msg="Recovered from panic in HTTP handler [86.229.216.31:52916 - /]: reflect.Set: value of type *http.Request is not assignable to type *struct { Transport http.RoundTripper; Provider interface {}; ConfigurationURL struct { TTL struct { Duration time.Duration } \"json:\\\"ttl\\\" yaml:\\\"ttl\\\"\"; Headers []string \"json:\\\"headers\\\" yaml:\\\"headers\\\"\" }; MarkCachedResponses bool; CoalescingLayerStorage *struct {}; YkeyStorage *struct { Cache *struct { Xcache *struct { XdefaultExpiration time.Duration; Xitems map[string]struct { Object interface {}; Expiration int64 }; Xmu sync.RWMutex; XonEvicted func(string, interface {}); Xjanitor *struct { Interval time.Duration; Xstop chan bool } } }; Keys map[string]struct { URL string \"json:\\\"url\\\" yaml:\\\"url\\\"\"; Headers map[string]string \"json:\\\"headers\\\" yaml:\\\"headers\\\"\" } } }" middlewareName=traefik-internal-recovery middlewareType=Recovery
https-proxy      | time="2021-09-06T19:01:01Z" level=error msg="Stack: goroutine 264 [running]:\ngithub.com/traefik/traefik/v2/pkg/middlewares/recovery.recoverFunc({0x7f64e3a0b8, 0x40034095c8}, 0x4001834000)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/recovery/recovery.go:46 +0x270\npanic({0x26613c0, 0x40000b7550})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4000845130, 0x400211b200, 0x4000a069f0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x26613c0, 0x40000b7550})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x400054ef20, 0x4000dce700, 0x4000a067d0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x26613c0, 0x40000b7550})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x400054f1e0, 0x400213e500, 0x4000a065b0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x26613c0, 0x40000b7550})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4001caa2c0, 0x4003b4f200, 0x4000a06390)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x26613c0, 0x40000b7550})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\ngithub.com/traefik/yaegi/interp.runCfg.func1(0x4001caafd0, 0x4003b35d00, 0x4000a06170)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:184 +0x168\npanic({0x26613c0, 0x40000b7550})\n\t/usr/local/golang/1.10.8/go/src/runtime/panic.go:1038 +0x224\nreflect.Value.assignTo({0x2cc6e80, 0x4003520338, 0x196}, {0x2dc7972, 0xb}, 0x4000d397c0, 0x0)\n\t/usr/local/golang/1.10.8/go/src/reflect/value.go:2797 +0x2cc\nreflect.Value.Set({0x4000d397c0, 0x4003520410, 0x196}, {0x2cc6e80, 0x4003520338, 0x196})\n\t/usr/local/golang/1.10.8/go/src/reflect/value.go:1905 +0xa4\ngithub.com/traefik/yaegi/interp.call.func6(0x4001caafd0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1272 +0xb38\ngithub.com/traefik/yaegi/interp.runCfg(0x4003b35d00, 0x4001caafd0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4001caa2c0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x4003b4f200, 0x4001caa2c0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x400054f1e0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x400213e500, 0x400054f1e0)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x400054ef20)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x4000dce700, 0x400054ef20)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.call.func6(0x4000845130)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:1283 +0x7ac\ngithub.com/traefik/yaegi/interp.runCfg(0x400211b200, 0x4000845130)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:190 +0x88\ngithub.com/traefik/yaegi/interp.genFunctionWrapper.func2.1({0x40045d20c0, 0x2, 0x2})\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/interp/run.go:951 +0x41c\ngithub.com/traefik/yaegi/stdlib._net_http_Handler.ServeHTTP(...)\n\t/home/semaphore/go/pkg/mod/github.com/traefik/yaegi@v0.9.23/stdlib/go1_17_net_http.go:269\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0x4002457100, {0x7f64e3a0b8, 0x40034095c8}, 0x4001834200)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:38 +0x130\ngithub.com/gorilla/mux.(*Router).ServeHTTP(0x4002a55920, {0x7f64e3a0b8, 0x40034095c8}, 0x4001834000)\n\t/home/semaphore/go/pkg/mod/github.com/containous/mux@v0.0.0-20181024131434-c33f32e26898/mux.go:133 +0x268\ngithub.com/traefik/traefik/v2/pkg/middlewares/recovery.(*recovery).ServeHTTP(0x40029bf8d0, {0x7f64e3a0b8, 0x40034095c8}, 0x4001834000)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/recovery/recovery.go:32 +0x8c\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.AddOriginFields({0x7f64e3a108, 0x40034095c0}, 0x4001834000, {0x33628e0, 0x40029bf8d0}, 0x40033e9280)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:55 +0x198\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0x4002457240, {0x7f64e3a108, 0x40034095c0}, 0x4001834000)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/field_middleware.go:36 +0x100\ngithub.com/traefik/traefik/v2/pkg/middlewares/requestdecorator.(*RequestDecorator).ServeHTTP(0x4001b95188, {0x7f64e3a108, 0x40034095c0}, 0x4003a09900, 0x4002bd3410)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/requestdecorator/request_decorator.go:47 +0x5e8\ngithub.com/traefik/traefik/v2/pkg/middlewares/requestdecorator.WrapHandler.func1.1({0x7f64e3a108, 0x40034095c0}, 0x4003a09900)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/requestdecorator/request_decorator.go:84 +0x70\nnet/http.HandlerFunc.ServeHTTP(0x40024372a0, {0x7f64e3a108, 0x40034095c0}, 0x4003a09900)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/middlewares/metrics.(*metricsMiddleware).ServeHTTP(0x4004247b90, {0x7f64e3a0b8, 0x40034095b8}, 0x4003a09900)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/metrics/metrics.go:121 +0x754\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*Handler).ServeHTTP(0x40014d78c0, {0x339f1d8, 0x40034095a0}, 0x4003a09800, {0x3362860, 0x4004247b90})\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/logger.go:227 +0x1110\ngithub.com/traefik/traefik/v2/pkg/middlewares/accesslog.WrapHandler.func1.1({0x339f1d8, 0x40034095a0}, 0x4003a09800)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/accesslog/logger.go:71 +0x50\nnet/http.HandlerFunc.ServeHTTP(0x40024372e0, {0x339f1d8, 0x40034095a0}, 0x4003a09800)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/server/router/tcp.(*Manager).buildEntryPointHandler.func1({0x339f1d8, 0x40034095a0}, 0x4003a09800)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/server/router/tcp/router.go:194 +0x188\nnet/http.HandlerFunc.ServeHTTP(0x40024374e0, {0x339f1d8, 0x40034095a0}, 0x4003a09800)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2046 +0x40\ngithub.com/traefik/traefik/v2/pkg/middlewares.(*HTTPHandlerSwitcher).ServeHTTP(0x400011a0f0, {0x339f1d8, 0x40034095a0}, 0x4003a09800)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/handler_switcher.go:23 +0x70\ngithub.com/traefik/traefik/v2/pkg/middlewares/forwardedheaders.(*XForwarded).ServeHTTP(0x400007e320, {0x339f1d8, 0x40034095a0}, 0x4003a09800)\n\t/home/semaphore/go/src/github.com/traefik/traefik/pkg/middlewares/forwardedheaders/forwarded_header.go:183 +0xec\nnet/http.serverHandler.ServeHTTP({0x40007f6000}, {0x339f1d8, 0x40034095a0}, 0x4003a09800)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:2878 +0x45c\nnet/http.initALPNRequest.ServeHTTP({{0x33c23e0, 0x40032b1650}, 0x4002bc9880, {0x40007f6000}}, {0x339f1d8, 0x40034095a0}, 0x4003a09800)\n\t/usr/local/golang/1.10.8/go/src/net/http/server.go:3479 +0x1b8\nnet/http.(*http2serverConn).runHandler(0x40017fe480, 0x40034095a0, 0x4003a09800, 0x40045c9e48)\n\t/usr/local/golang/1.10.8/go/src/net/http/h2_bundle.go:5823 +0x94\ncreated by net/http.(*http2serverConn).processHeaders\n\t/usr/local/golang/1.10.8/go/src/net/http/h2_bundle.go:5553 +0x524\n" middlewareName=traefik-internal-recovery middlewareType=Recovery
```

--- a/plugins/traefik/go.mod
+++ b/plugins/traefik/go.mod
@@ -8,3 +8,5 @@ require (
 	go.uber.org/zap v1.19.0
 	golang.org/x/sys v0.0.0-20210809222454-d867a43fc93e // indirect
 )
+
+replace github.com/darkweak/souin v1.5.7 => ../..

--- a/rfc/bridge.go
+++ b/rfc/bridge.go
@@ -34,13 +34,13 @@ func CachedResponse(c types.AbstractProviderInterface, req *http.Request, cached
 	}, nil
 }
 
-func commonCacheControl(req *http.Request, t func(*http.Request) (*http.Response, error)) (*http.Response, error) {
+func commonCacheControl(req *http.Request, t http.RoundTripper) (*http.Response, error) {
 	reqCacheControl := parseCacheControl(req.Header)
 	if _, ok := reqCacheControl["only-if-cached"]; ok {
 		return newGatewayTimeoutResponse(req), nil
 	}
 
-	r, err := t(req)
+	r, err := t.RoundTrip(req)
 
 	if err != nil {
 		return nil, err
@@ -122,12 +122,16 @@ func (t *VaryTransport) UpdateCacheEventually(req *http.Request) (*http.Response
 	if cacheable && cachedResp != nil {
 		rDate, _ := time.Parse(time.RFC1123, req.Header.Get("Date"))
 		cachedResp.Header.Set("Date", rDate.Format(http.TimeFormat))
+		cachedResp.Header.Set("Cache-Status", "Souin; fwd=uri-miss: stored")
 		r := commonVaryMatchesVerification(cachedResp, req)
 		if r != nil {
 			return r, nil
 		}
 	} else {
-		if _, err := commonCacheControl(req, t.RoundTrip); err != nil {
+		if cachedResp != nil {
+			cachedResp.Header.Set("Cache-Status", "Souin; fwd=uri-miss")
+		}
+		if _, err := commonCacheControl(req, t); err != nil {
 			return nil, err
 		}
 	}
@@ -185,7 +189,7 @@ func (t *VaryTransport) RoundTrip(req *http.Request) (resp *http.Response, err e
 			}
 		}
 	} else {
-		if resp, err = commonCacheControl(req, transport.RoundTrip); err != nil {
+		if resp, err = commonCacheControl(req, transport); err != nil {
 			return nil, err
 		}
 		resp.Header.Set("Cache-Status", "Souin; fwd=uri-miss")

